src/corelibs/U2Core/src/datatype/msa/MultipleAlignment.cpp

```diff
@@ -178,14 +168,46 @@ static bool isLessByLength(const MultipleAlignmentRow &row1, const MultipleAlign
     return row1->getUngappedLength() < row2->getUngappedLength();
 }
 
-void MultipleAlignmentData::sortRowsByLength(MultipleAlignment::Order order, const U2Region &range) {
+/** Returns leading gap size of the given row. */
+static int getLeadingGapSize(const MultipleAlignmentRow &row) {
+    const U2MsaRowGapModel &gapModel = row->getGapModel();
+    if (gapModel.isEmpty()) {
+        return 0;
+    }
+    const U2MsaGap &gap = gapModel.first();
+    return gap.offset == 0 ? gap.gap : 0;
+}
```

Comment:
Move it to MultipleAlignmentRowData class.

Reply:
Fixed